core/templates/dev/head/services/contextual/UrlService.js

```diff
@@ -19,25 +19,41 @@
 
 oppia.factory('UrlService', ['$window', function($window) {
   return {
+    // This function is for testing purposes (to mock $window.location)
+    getCurrentUrl: function() {
+      return $window.location;
+    },
     getUrlParams: function() {
       var params = {};
-      var parts = $window.location.href.replace(
+      var parts = this.getCurrentUrl().href.replace(
         /[?&]+([^=&]+)=([^&]*)/gi, function(m, key, value) {
           params[key] = value;
         }
       );
       return params;
     },
+    getParentExplorationId: function() {
+      var params = {};
+      var parts = this.getCurrentUrl().href.replace(
```

Comment:
Can this be replaced with this.getParameters()?

Reply:
I removed that function altogether, was not needed with the new method.